theano/sandbox/cuda/basic_ops.py

```diff
@@ -3044,6 +3044,24 @@ def make_node(self, *axis_and_tensors):
         as_tensor_variable_args = [as_cuda_ndarray_variable(x)
                                    for x in tensors]
 
+        # If axis is negative, we need to increment it with the number
+        # of dimensions of the smallest tensor variable
+        axis_int = int(axis.eval())
```

Comment:
As far as I know, all tensor must have the same dimension. See [here](https://github.com/Theano/Theano/blob/master/theano/tensor/basic.py#L3473) in the CPU version. 

You could simply take first tensor dimension for setting the positive axis value. No need to check that all tensor have same dimension has it's done in ```tensor.Join._make_node_internal()``` just below.